src/measurement.rs

```diff
@@ -0,0 +1,163 @@
+use bitfield::{bitfield, Bit, BitRange};
+
+#[derive(Copy, Clone, Default)]
+pub struct U12(u16);
+
+impl U12 {
+    pub fn new(value: u16) -> U12 {
+        U12(value.min(0xFFF))
+    }
```

Comment:
One last detail: This can be simplified.

```suggestion
    pub fn new(value: u16) -> Self {
        Self(value.min(0xFFF))
    }
```

Reply:
Why is this simpler, it needs two chars more :laughing:

Reply:
Fixed :slightly_smiling_face: